src/node/node_state.h

```diff
@@ -596,6 +599,282 @@ namespace ccf
         std::chrono::milliseconds(config.joining.join_timer));
     }
 
+    template <typename T>
```

Comment:
Can we move this new code to a different file? We should aim to have `node_state.h` only deal with the internal state of the node.

Reply:
Sure, I identified the following things that I use from the node state instance:
- `std::lock_guard<SpinLock> guard(lock);`
- `rpcsessions->create_client`
- `network.tables->create_read_only_tx();`

I would leave the `auto_refresh_jwt_keys` method in node_state and move the rest out, passing rpcsessions and tx around. The only thing I'm not sure about is the lock. When do I need that?

Reply:
The `lock` is to protect the node's state so as long as you don't touch those (which I believe you don't), you should be fine to move this to a new class/file.

Reply:
Some more thing:
- `node_cert`
- `rpc_map`